packages/adminrouter/extra/src/test-harness/tests/test_boot_envvars.py

```diff
@@ -16,73 +17,277 @@ def empty_file(tmp_file):
 
         assert resp.status_code == 401
+
+
+class TestDefaultSchemeEnvVarBehaviour():
+    def test_if_default_scheme_is_honourded_by_agent_endpoint(
+            self, nginx_class, mocker, superuser_user_header):
+        filter_regexp = 'Default scheme: https://'
+
+        ar = nginx_class(default_scheme="https://")
+        agent_id = '35f210bb-bb58-4559-9932-b62619e72b6d-S0'
+        url_good = ar.make_url_from_path('/agent/{}/blah/blah'.format(agent_id))
+        agent_id = 'de1baf83-c36c-4d23-9cb0-f89f596cd6ab-S1'
+        url_bad = ar.make_url_from_path('/agent/{}/blah/blah'.format(agent_id))
+
+        with GuardedSubprocess(ar):
+            lbf = LineBufferFilter(filter_regexp,
+                                   line_buffer=ar.stderr_line_buffer)
+
+            resp = requests.get(url_bad,
+                                allow_redirects=False,
+                                headers=superuser_user_header)
+
+            assert resp.status_code == 502
+
+            resp = requests.get(url_good,
+                                allow_redirects=False,
+                                headers=superuser_user_header)
+
+            assert resp.status_code == 200
+            req_data = resp.json()
+            assert req_data['endpoint_id'] == 'https://127.0.0.1:15401'
+
+            lbf.scan_log_buffer()
+
+        assert lbf.all_found is True
+
+    def test_if_default_scheme_is_honourded_by_mleader_endpoint(
+            self, nginx_class, mocker, superuser_user_header):
+        filter_regexp = 'Default scheme: https://'
+
+        cache_poll_period = 3
+        ar = nginx_class(cache_poll_period=cache_poll_period,
+                         cache_expiration=cache_poll_period - 1,
+                         default_scheme="https://")
+        url = ar.make_url_from_path('/system/v1/leader/marathon/foo/bar/baz')
+
+        with GuardedSubprocess(ar):
+            lbf = LineBufferFilter(filter_regexp,
+                                   line_buffer=ar.stderr_line_buffer)
+            resp = requests.get(url,
+                                allow_redirects=False,
+                                headers=superuser_user_header)
+
+            assert resp.status_code == 502
+
+            mocker.send_command(endpoint_id='http://127.0.0.1:8080',
+                                func_name='change_leader',
+                                aux_data="127.0.0.4:443")
+
+            # First poll (2s) + normal poll interval(4s) < 2 * normal poll
+            # interval(4s)
+            time.sleep(cache_poll_period * 2)
+
+            resp = requests.get(url,
+                                allow_redirects=False,
+                                headers=superuser_user_header)
+
+            assert resp.status_code == 200
+            req_data = resp.json()
+            assert req_data['endpoint_id'] == 'https://127.0.0.4:443'
+
+            lbf.scan_log_buffer()
+
+        assert lbf.all_found is True
```

Comment:
Yes, and this is a perfect example why :)

Even though I wrote this test harness, even though I know that `all_found` is inferior to `extra_matches`, I still copypasted it :/ I will remove `all_found` from `LineBufferFilter` so that there is only one, obvious way to do matching. Same goes for regexp filter specification.